src/main/java/org/spongepowered/api/service/permission/PermissionDescription.java

```diff
@@ -231,6 +283,25 @@
          */
         Builder assign(String role, boolean value);
 
+
+        /**
+         * Sets a value that this permission should have by default.
+         * This can be used to exclude permissions from node tree inheritance,
+         * or to provide a permission to users by default.
+         *
+         * <p>This is shorthand for giving {@link #getId()} (with templates stripped) a value on the default
+         * subject, except that the default value will only be applied once
+         * {@link #register()} is called.</p>
+         *
+         * <p>Assigning default permissions should be used sparingly, and by
+         * convention, only in situations where "default" game behaviour is restored
+         * by granting a certain permission.</p>
+         *
+         * @param defaultValue The value this permission should have for subjects where none has been assigned.
+         * @return The builder for chaining.
+         */
+        Builder defaultValue(Tristate defaultValue);
```

Comment:
Sooo, is the permissions implementation required to take this into account when resolving permission checks?

It was my understanding that the default subject / subjectcollection was the way to do this - it seems unnecessary to duplicate that functionality.

Reply:
It was my impression that permission descriptions could become more like a "template" that a plugin can supply and a server admin can apply - basically so that `nucleus setupperms` can go the way of the dodo. I'm not sure how this is intended, whether templates can act as immutable groups or the permissions just get copied across to the group. I would not want templates to be applied by default, however.

On this specifically, the intention is that a default that some templates might want to set for permissions is explicit false.

Reply:
I wished, that plugins would not mess with the default subject either.
The plugins should only define the descriptions and suggest specific values for certain ranks.
If no perm plugin is present, then Sponge's PermissionService should only check the `*:USER` (or `*:DEFAULT`) rank (or a list of configured ranks) and take that value and else depend on isOP().
If a perm plugin is present, then it might initially setup itself in a similar fashion, but if the admin removes a rank/permission from the default, then it should be gone for good. Plugins shouldn't sneakily add it back (at least not as the recommended way). The server admin added the permission plugin for a reason.

Reply:
This is designed as a helper method that delegates to the default subject -- so a `PermissionDescription` does not hold that data itself -- it just provides convenient access so all the plugin developer-focused metadata about a permission can be set up in one place.

The reason the default subject is exposed in the API is so that plugin developers can register permissions that should be granted by default on its transient data. This has been used since the beginning by SpongeCommon itself for permissions registrations. It's up to the specific permissions plugin to let users override -- for example PEX makes the persistent data take priority over the transient data for the default subject.

in the event no permissions plugin is present, spongecommon already performs a simplified version of permissions resolution logic based on transient data and the user's op level, and that has not changed since the api was originally implemented.